data/transactions/logic/pairing_test.go

```diff
@@ -16,14 +16,543 @@
 
 package logic
 
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/algorand/go-algorand/test/partitiontest"
+	bls12381 "github.com/consensys/gnark-crypto/ecc/bls12-381"
+	bls12381fp "github.com/consensys/gnark-crypto/ecc/bls12-381/fp"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+	bn254fp "github.com/consensys/gnark-crypto/ecc/bn254/fp"
+)
+
 const pairingNonsense = `
  pushbytes 0x012345
  dup
- bn256_add
+ ec_add BN254g1
  dup
- bn256_scalar_mul
+ ec_scalar_mul BLS12_381g2
  dup
- bn256_pairing
+ ec_pairing_check BN254g1
+ ec_multi_exp BLS12_381g2
+ ec_subgroup_check BLS12_381g1
+ ec_map_to BN254g2
 `
 
-const pairingCompiled = "80030123454999499a499b"
+const pairingCompiled = "800301234549e00049e10349e200e303e402e501"
+
+func bn254G1sToBytes(g1s []bn254.G1Affine) []byte {
+	var out []byte
+	for i := range g1s {
+		out = append(out, bn254G1ToBytes(&g1s[i])...)
+	}
+	return out
+}
+
+func bn254G2sToBytes(g2s []bn254.G2Affine) []byte {
+	var out []byte
+	for i := range g2s {
+		out = append(out, bn254G2ToBytes(&g2s[i])...)
+	}
+	return out
+}
+
+func bls12381G1sToBytes(g1s []bls12381.G1Affine) []byte {
+	var out []byte
+	for i := range g1s {
+		out = append(out, bls12381G1ToBytes(&g1s[i])...)
+	}
+	return out
+}
+
+func bls12381G2sToBytes(g2s []bls12381.G2Affine) []byte {
+	var out []byte
+	for i := range g2s {
+		out = append(out, bls12381G2ToBytes(&g2s[i])...)
+	}
+	return out
+}
+
+func TestEcAdd(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	// Parameterize this to check all curves
+
+	g1point := bn254RandomG1()
+	g1bytes1 := fmt.Sprintf("byte 0x%s\n", hex.EncodeToString(bn254G1ToBytes(&g1point)))
+	g1point = bn254RandomG1()
+	g1bytes2 := fmt.Sprintf("byte 0x%s\n", hex.EncodeToString(bn254G1ToBytes(&g1point)))
```

Comment:
It might be somewhere else, but I think it's important to test also all the corner cases:
* wrong length for first point
* wrong length for second point
* right length but not on curve for first point
* same for second point
* addition with first point = point to infinity (done already)
* addition with second point = point to infinity
* addition with both point of infinity
* addition of points that are on the curve but not in the subgroup.